lib/rabbit_feed/event_definitions.rb

```diff
@@ -51,19 +51,31 @@ def defined_as &block
         @definition = block.call if block.present?
       end
 
-      def payload
-        ([
+      def payload_schema
+        { name: "#{name}_payload", type: 'record', fields: fields.map(&:schema) }
+      end
+
+      def metadata_schema
+        { name: 'event_metadata', type: 'record', fields: [
           (Field.new 'application',    'string', 'The name of the application that created the event'),
           (Field.new 'host',           'string', 'The hostname of the server on which the event was created'),
           (Field.new 'environment',    'string', 'The environment in which the event was created'),
-          (Field.new 'version',        'string', 'The version of the event'),
+          (Field.new 'version',        'string', 'The version of the event payload'),
+          (Field.new 'gem_version',    'string', 'The version of rabbit_feed by which the event was created'),
```

Comment:
Not super strong at the moment... but I feel it can cause issues. For example if we do separate the eventing part from the Rabbit gem, that will mean that I will have to start with version 2.0.0 of that EventWhatever gem. which would seem kind of strange. While starting with version 2 of the events doesn't seem as strange because they are the same events just moved elsewhere...